ako-gateway-api/nodes/gateway_model_rel.go

```diff
@@ -259,17 +267,22 @@ func HTTPRouteToGateway(namespace, name, key string) ([]string, bool) {
 
 					gwListenerNsName := gwNsName + "/" + listener.Name
 					listenerHostname := akogatewayapiobjects.GatewayApiLister().GetGatewayListenerToHostname(gwListenerNsName)
-					if strings.HasPrefix(listenerHostname, "*") {
-						listenerHostname = listenerHostname[1:]
-					}
+
+					listenerHostname = strings.TrimPrefix(listenerHostname, utils.WILDCARD)
+
 					hostnameMatched := false
 					for _, routeHostname := range hrObj.Spec.Hostnames {
-						if strings.HasSuffix(string(routeHostname), listenerHostname) && akogatewayapilib.VerifyHostnameSubdomainMatch(string(routeHostname)) {
-							hostnameIntersection = append(hostnameIntersection, string(routeHostname))
-							hostnameMatched = true
+						// When Gateway hostname is empty, then just check validity of hostname and append it.
+						// When hostname in HTTProute has wildcard
+						// When there is exact match
+						if strings.HasSuffix(string(routeHostname), listenerHostname) || listenerHostname == "" || strings.HasPrefix(string(routeHostname), utils.WILDCARD) {
```

Comment:
Will this condition not lead to true even if listener hostname is a.b.c and routehostname is d.a.b.c ?
both of these hostnames are neither overlapping nor exact match